text/0000-language-semver.md

```diff
@@ -0,0 +1,494 @@
+intentionally unspecified. One obvious example is the layout of data
+structures, which is considered undefined unless they have a
+`#[repr(C)]` attribute.
+
+Although it is not directly covered by this RFC, it's worth noting in
+passing that some of the CLI flags to the compiler may change in the
+future as well. The `-Z` flags are of course explicitly unstable, but
+some of the `-C`, rustdoc, and linker-specific flags are expected to
+evolve over time.
+
+### Opt-in changes
+
+For breaking changes that are not related to soundness or language
+semantics, but are still deemed desirable, an opt-in strategy can be
+used instead. This section describes an attribute for opting in to
+newer language updates, and gives guidelines on what kinds of changes
+should or should not be introduced in this fashion.
+
+We use the term *"opt-in changes"* to refer to changes that would be
+breaking changes, but are not because of the opt-in mechanism.
+
+#### Rust version option
+
+The specific proposal is to introduce a command-line option
+`--rust-version=X.Y[.Z]` that instructs the Rust compiler to expect
+source code from older versions of Rust. This option could also be
+specified in a `Cargo.toml` file in a `rust-version` property.  The
+version applies to the crate currently being compiled and is called
+the crate's "supplied version". Every build of the Rust compiler will
+also have a version number built into it reflecting the current
+release; if the command-line option is not supplied, the compiler
+defaults to this builtin version.
+
+The supplied version is used by the compiler to produce the semantics
+of Rust "as it was" during version `X.Y`. RFCs that propose opt-in
+changes should discuss how the older behavior can be supported in the
+compiler, but this is expected to be straightforward: if supporting
+older behavior is hard to do, it may indicate that the opt-in change
+is too complex and should not be accepted.
+
+Note that the supplied version may affect the parser configuration
+used when parsing the initial crate, since it can affect the keywords
+recognized by the tokenizer and perhaps other minor details in the
+syntax. However, because the version is supplied on the command line,
+this configuration is known before parsing begins.
+
+#### Defaults and extreme cases
+
+If no version is supplied on the `rustc` command line, `rustc` will
+default to the maximal version it recognizes. If the user supplies a
+version `X.Y` that is *newer* than the compiler itself, the compiler
+should simply issue a warning and proceed as if the user had supplied
+the compiler's version (i.e., the newest version the compiler knows
+about).
+
+Cargo will always invoke `rustc` with a supplied version. If there is
+no version in the `Cargo.toml` file, then `1.0.0` is assumed. Whenever
+a new project is created with `cargo new`, the new `Cargo.toml` will
+include the most recent Rust version number by default.
```

Comment:
since Cargo and Rust are not developed nor released synchronously, the phrase "most recent Rust version number" with respect to Cargo may need some clarification.  Does it mean:

 * The most recent Rust version at the time of that Cargo release.
 * The Rust version that the installed Cargo binary observes (presumably via the `$PATH` env var, but perhaps there will also be some configuration to point Cargo at specific Rust installations at some point, much like `multirust` attempts to support today...)
 * The most recent Rust version that Cargo can find by querying the web site?

(I know which one of the above options is my guess for what was intended, but maybe there's even something else entirely that you meant...)